src/utils/AnalyticsUtils.ts

```diff
@@ -0,0 +1,13 @@
+import { IAnalyticsActionCause, analyticsActionCauseList } from '../ui/Analytics/AnalyticsActionListMeta';
+
+export class AnalyticsUtils {
+  static addActionCauseToList(newActionCause: IAnalyticsActionCause) {
+    if (newActionCause.name && newActionCause.type) {
+      analyticsActionCauseList[newActionCause.name] = newActionCause;
```

Comment:
Do we want to override existing action causes too, or do we want to prevent it?

Reply:
Hmm I'm not sure. Would there be benefits to prevent updating?